packages/modeling/src/utils/uniquePoints.js

```diff
@@ -0,0 +1,33 @@
+const geom2 = require("../geometries/geom2");
+const geom3 = require("../geometries/geom3");
+const path2 = require("../geometries/path2");
+
+/*
+ * Return the unique vertices of a geometry
+ */
+const uniquePoints = (geometries) => {
+  const found = new Set()
+  const uniquePoints = []
+
+  const addPoint = (point) => {
+    const key = point.toString()
+    if (!found.has(key)) {
+      uniquePoints.push(point)
+      found.add(key)
+    }
+  }
+
+  geometries.forEach((geometry) => {
+    if (geom2.isA(geometry)) {
+      geom2.toPoints(geometry).forEach(addPoint)
+    } else if (geom3.isA(geometry)) {
+      geom3.toPoints(geometry).forEach(addPoint)
+    } else if (path2.isA(geometry)) {
+      path2.toPoints(geometry).forEach(addPoint)
+    }
+  })
+
```

Comment:
Does it make sense to call found.clear() to release those strings back to garbage cleaning? I haven't seen any documentation on the benefits of clear() so just wondering.

Reply:
The Set goes out of scope as soon as the function returns, allowing the keys to be freed. There should be no need to call clear().